src/hotspot/share/jfr/metadata/metadata.xml

```diff
@@ -706,6 +706,19 @@
     <Field type="OldObjectGcRoot" name="root" label="GC Root" />
   </Event>
 
+  <Event name="NativeMemoryUsagePart" category="Java Virtual Machine, Memory" label="Component Native Memory Usage" description="Native memory usage for a component" stackTrace="false" thread="false"
+    startTime="false" period="everyChunk">
+    <Field type="string" name="type" label="Memory Type" description="Component allocating the native memory" />
```

Comment:
There is a way to create special JFR types and instead just send an ID. This might make sense to save some extra memory. @egahlin, is there any drawback with creating custom type? Apart from the parsing needing an extra step.

Reply:
Not really. See for https://github.com/openjdk/jdk/blob/master/src/hotspot/share/jfr/metadata/jfrSerializer.hpp for how to serialize constants. Since the number of events are few and they are opt-in using a command line flag, I don't think it will make much of a difference from a performance perspective, but I don't think it will hurt either.